flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -169,6 +169,16 @@ public ExecutionVertexInputInfo getExecutionVertexInputInfo(IntermediateDataSetI
                 .get(subTaskIndex);
     }
 
+    /** Get the current input number bytes. Note that only finished inputs will be counted. */
+    public long getExecutionVertexInputNumBytes() {
+        return getJobVertex().getInputs().stream()
+                .mapToLong(
+                        intermediateResult ->
+                                getExecutionVertexInputInfo(intermediateResult.getId())
+                                        .getAggregatedInputNumBytes())
+                .sum();
```

Comment:
It's better to record the total input bytes and directly return it, to avoid heavy invocations per-second(slow task detecting). The total input bytes can be computed when the execution vertex is scheduled.

Then we do not need to introduce a `aggregatedInputNumBytes` to `ExecutionVertexInputInfo`.

Reply:
Yes, it's better, fix this.